.travis.yml

```diff
@@ -1,51 +1,114 @@
+# This .travis.yml file was autogenerated from github.com/ericdill/travis-little-helper/template.py.
+# Please do not edit this by hand!
+
 language: python
+
 sudo: false
+
 services:
-- mongodb
+  - mongodb
 matrix:
+  fast_finish: true
   include:
   - python: 2.7
+    env: PYMONGO="<3"
+  - python: 2.7
+    env: PYMONGO=">3"
+  - python: 3.4
+    env: PYMONGO="<3"
   - python: 3.4
+    env: PYMONGO=">3"
+  - python: 3.5
+    env: PYMONGO="<3"
   - python: 3.5
+    env: PYMONGO=">3"
+  - python: nightly
+    env: PYMONGO="<3"
+  - python: nightly
+    env: PYMONGO=">3"
+
+  allow_failures:
+    - python: nightly
+    - env: PYMONGO=">3"
+
+
 before_install:
-- wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
-- chmod +x miniconda.sh
-- "./miniconda.sh -b -p /home/travis/mc"
-- export PATH=/home/travis/mc/bin:$PATH
-- export MDS_HOST=localhost
-- export MDS_DATABASE=test
-- export MDS_TIMEZONE=US/Eastern
-- mkdir -p /home/travis/.config/metadatastore
-- 'echo ''port: 27017'' > /home/travis/.config/metadatastore/connection.yml'
+  - env
+  # `git describe` failed a couple of times if I don't `git fetch --unshallow`
+  - git fetch --unshallow
+  # GRAB THE HELPER SCRIPTS
+  - git clone https://github.com/ericdill/travis-little-helper ../travis-little-helper
+  # ECHO THE ENVIRONMENTAL VARIABLES THAT WERE SET
+  
+  - echo $PYMONGO
+  # INSTALL CONDA
+  - wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
+  - chmod +x miniconda.sh
+  - "./miniconda.sh -b -p /home/travis/mc"
+  - export PATH=/home/travis/mc/bin:$PATH
+  - conda config --set always_yes true
+  - conda update conda --yes
+  - conda config --add channels lightsource2
+  - conda config --add channels lightsource2-dev
+  
+  # DO SOME METADATASTORE SPECIFIC CONFIGURATION
+  - export MDS_HOST=localhost
+  - export MDS_DATABASE=test
+  - export MDS_TIMEZONE=US/Eastern
+  - mkdir -p /home/travis/.config/metadatastore
+  - 'echo ''port: 27017'' > /home/travis/.config/metadatastore/connection.yml'
+  
+  # CONFIGURE USEFUL ENVIRONMENTAL VARIABLES
+  - export OWNER_NAME=$(IFS="/"; arr=($TRAVIS_REPO_SLUG); echo ${arr[0]})
+  - export REPO_NAME=$(IFS="/"; arr=($TRAVIS_REPO_SLUG); echo ${arr[1]})
+  - export CONDA_BUILD_COMMAND="conda build conda-recipe --python=$TRAVIS_PYTHON_VERSION"
+  - export GIT_FULL_HASH=`git rev-parse HEAD`
+  - export CLEAN_ANACONDA_CHANNEL=False
+  # SOME DEBUG OUTPUT
+  - echo "OWNER_NAME=$OWNER_NAME"
+  - echo "REPO_NAME=$REPO_NAME"
+  - echo "TRAVIS_COMMIT=$TRAVIS_COMMIT"
+  - git describe
+  # INSTALL DEPS FOR BUILDING AND UPLOADING
+  - conda install conda-build jinja2 anaconda-client
```

Comment:
yeah i saw that PR finally get merged. I agree that this should be left for now until we are sure that the most recent version of conda-build will install jinja2